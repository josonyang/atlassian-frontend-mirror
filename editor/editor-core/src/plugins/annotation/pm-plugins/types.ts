import type { DecorationSet } from '@atlaskit/editor-prosemirror/view';
import type {
  EditorState,
  SelectionBookmark,
} from '@atlaskit/editor-prosemirror/state';
import type { PortalProviderAPI } from '@atlaskit/editor-common/portal-provider';
import type {
  Dispatch,
  EventDispatcher,
} from '@atlaskit/editor-common/event-dispatcher';
import type { InlineCommentAnnotationProvider, AnnotationInfo } from '../types';

export enum ACTIONS {
  UPDATE_INLINE_COMMENT_STATE,
  SET_INLINE_COMMENT_DRAFT_STATE,
  INLINE_COMMENT_UPDATE_MOUSE_STATE,
  INLINE_COMMENT_CLEAR_DIRTY_MARK,
  ADD_INLINE_COMMENT,
  INLINE_COMMENT_SET_VISIBLE,
  CLOSE_COMPONENT,
  SET_SELECTED_ANNOTATION,
}

export interface InlineCommentPluginOptions {
  dispatch: Dispatch;
  eventDispatcher: EventDispatcher;
  portalProviderAPI: PortalProviderAPI;
  provider: InlineCommentAnnotationProvider;
}
export interface InlineCommentMouseData {
  isSelecting: boolean;
}

export type InlineCommentMap = { [key: string]: boolean };
export type InlineCommentAction =
  | {
      type: ACTIONS.UPDATE_INLINE_COMMENT_STATE;
      data: InlineCommentMap;
    }
  | {
      type: ACTIONS.SET_INLINE_COMMENT_DRAFT_STATE;
      data: {
        drafting: boolean;
        editorState: EditorState;
      };
    }
  | {
      type: ACTIONS.INLINE_COMMENT_UPDATE_MOUSE_STATE;
      data: {
        mouseData: InlineCommentMouseData;
      };
    }
  | { type: ACTIONS.INLINE_COMMENT_CLEAR_DIRTY_MARK }
  | {
      type: ACTIONS.CLOSE_COMPONENT;
      data: {
        lastClosedPos: number;
      };
    }
  | {
      type: ACTIONS.ADD_INLINE_COMMENT;
      data: {
        drafting: boolean;
        inlineComments: InlineCommentMap;
        editorState: EditorState;
        selectedAnnotations: AnnotationInfo[];
      };
    }
  | { type: ACTIONS.INLINE_COMMENT_SET_VISIBLE; data: { isVisible: boolean } }
  | {
      type: ACTIONS.SET_SELECTED_ANNOTATION;
      data: { selectedAnnotations: AnnotationInfo[] };
    };

export type InlineCommentPluginState = {
  annotations: InlineCommentMap;
  selectedAnnotations: AnnotationInfo[];
  dirtyAnnotations?: boolean;
  mouseData: InlineCommentMouseData;
  draftDecorationSet?: DecorationSet;
  bookmark?: SelectionBookmark;

  // Denotes if annotations are allowed to be create on empty nodes or nodes of whitespace (Confluence spec)
  disallowOnWhitespace: boolean;

  // Allow users to hide inline comments during editing
  isVisible: boolean;

  // Position of the last manually closed comment
  // This is used to prevent the closed comment from reopening unintentionally
  lastClosedPos?: number;
};
