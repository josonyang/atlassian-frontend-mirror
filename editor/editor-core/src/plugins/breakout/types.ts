import { ContentNodeWithPos } from 'prosemirror-utils';
export interface BreakoutPluginState {
  breakoutNode: ContentNodeWithPos;
}
