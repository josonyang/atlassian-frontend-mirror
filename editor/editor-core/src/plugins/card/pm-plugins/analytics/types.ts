import { Mark, Node } from 'prosemirror-model';

export type Link =
  | {
      type: 'node';
      pos: number;
      node: Node;
    }
  | {
      type: 'mark';
      pos: number;
      mark: Mark;
    };
