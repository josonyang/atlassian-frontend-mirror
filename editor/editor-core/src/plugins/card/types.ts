import { INPUT_METHOD, ACTION } from '@atlaskit/editor-common/analytics';
import {
  CardProvider,
  CardAppearance,
} from '@atlaskit/editor-common/provider-factory';
import { SmartLinkEvents } from '@atlaskit/smart-card';
import {
  CreateUIAnalyticsEvent,
  UIAnalyticsEvent,
} from '@atlaskit/analytics-next';
import { CardOptions } from '@atlaskit/editor-common/card';
import type { EditorAppearance } from '@atlaskit/editor-common/types';
import { LinkPickerOptions } from '@atlaskit/editor-common/types';

export type CardInfo = {
  title?: string;
  url?: string;
  pos: number;
};

export type Request = {
  /**
   * The position of the link in the doc this request for upgrade is for
   */
  pos: number;
  /**
   * The URL of the link being upgraded/resolved
   */
  url: string;
  /**
   * The requested appearance to upgrade to
   */
  appearance: CardAppearance;
  /**
   * Determines if the link should only be replaced after successful
   * resolution if the link's text content still matches its href
   * (ie has not been changed by the user since the resolve request).
   */
  compareLinkText: boolean;
  /**
   * Analytics input method which preserves "how" the change
   * has occured. Used when analytics are dispatched after the link
   * is successfully/unsuccessfully resolved.
   */
  source: CardReplacementInputMethod;
  /**
   * The previous appearance of the link
   * This is necessary when tracking an update of a link
   * where the link begins as a smart card, then is updated by
   * replacing with a hyperlink and queueing it for upgrade
   */
  previousAppearance?: CardAppearance | 'url';
  /**
   * Analytics action to be used when disaptching
   * analytics events once the link is successfully/unsuccessfully
   * resolved. Preserves the users "intent" ie was this a link
   * being inserted or updated?
   */
  analyticsAction?: ACTION;
  /**
   * Describes if the requested `appearance` MUST be returned by the card provider
   * when resolving the link, because the request is associated with the user intent
   * to explicitly change the link appearance to the target appearance (view switcher), and therefore
   * should not be affected by "default appearances" or "user preferences"
   */
  shouldReplaceLink?: boolean;
  /**
   * Source UI that triggered this step if relevant
   * Primarily for transfering link picker UI event from initial UI action
   * through to the end of the link queue/request saga
   */
  sourceEvent?: UIAnalyticsEvent | null | undefined;
};

type Metadata<T = {}> = {
  url: string;
  display: string;
  isUndo?: boolean;
  isRedo?: boolean;
  action?: string;
  inputMethod?: string;
  sourceEvent?: unknown;
  nodeContext?: string;
} & T;

type UpdateMetadata = {
  previousDisplay?: string;
};

export type SmartLinkEventsNext = {
  created: (metadata: Metadata) => void;
  updated: (metadata: Metadata<UpdateMetadata>) => void;
  deleted: (metadata: Metadata) => void;
};

export type LifecycleEventType = keyof SmartLinkEventsNext;

/**
 * Describes the shape of an event that will be stored
 * in the Card state until it can be dispatched
 * as a side-effect in a view update
 */
export type LifecycleEvent<
  Type extends keyof SmartLinkEventsNext = keyof SmartLinkEventsNext,
> = {
  type: Type;
  data: Parameters<SmartLinkEventsNext[Type]>[0];
};

export type CardPluginState = {
  requests: Request[];
  provider: CardProvider | null;
  cards: CardInfo[];
  showLinkingToolbar: boolean;
  smartLinkEvents?: SmartLinkEvents;
  smartLinkEventsNext?: SmartLinkEventsNext;
  createAnalyticsEvent?: CreateUIAnalyticsEvent;
  editorAppearance?: EditorAppearance;
};

export type CardPluginOptions = CardOptions & {
  editorAppearance?: EditorAppearance;
  platform: 'mobile' | 'web';
  fullWidthMode?: boolean;
  createAnalyticsEvent?: CreateUIAnalyticsEvent;
  linkPicker?: LinkPickerOptions;
};

// actions
export type SetProvider = {
  type: 'SET_PROVIDER';
  provider: CardProvider | null;
};

export type Queue = {
  type: 'QUEUE';
  requests: Request[];
};

export type Resolve = {
  type: 'RESOLVE';
  url: string;
};

export type Register = {
  type: 'REGISTER';
  info: CardInfo;
};

export type ShowLinkToolbar = {
  type: 'SHOW_LINK_TOOLBAR';
};

export type HideLinkToolbar = {
  type: 'HIDE_LINK_TOOLBAR';
};

export type RegisterSmartCardEvents = {
  type: 'REGISTER_EVENTS';
  smartLinkEvents: SmartLinkEvents;
};

export type RegisterSmartCardEventsNext = {
  type: 'REGISTER_EVENTS_NEXT';
  smartLinkEvents: SmartLinkEventsNext;
};

export type CardPluginAction =
  | SetProvider
  | Queue
  | Resolve
  | Register
  | ShowLinkToolbar
  | HideLinkToolbar
  | RegisterSmartCardEvents
  | RegisterSmartCardEventsNext;

export type CardReplacementInputMethod =
  | INPUT_METHOD.CLIPBOARD
  | INPUT_METHOD.AUTO_DETECT
  | INPUT_METHOD.FORMATTING
  | INPUT_METHOD.MANUAL
  | INPUT_METHOD.TYPEAHEAD
  | INPUT_METHOD.FLOATING_TB;
