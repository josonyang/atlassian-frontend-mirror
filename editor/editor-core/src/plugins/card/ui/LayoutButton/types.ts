import { EditorView } from 'prosemirror-view';
import { IntlShape } from 'react-intl-next';
import type cardPlugin from '../../index';
import { ExtractInjectionAPI } from '@atlaskit/editor-common/types';

export type DatasourceTableLayout = 'full-width' | 'center' | 'wide';

export type LayoutButtonProps = {
  mountPoint?: HTMLElement;
  boundariesElement?: HTMLElement;
  scrollableElement?: HTMLElement;
  targetElement?: HTMLElement;
  layout: DatasourceTableLayout;
  onLayoutChange?: (layout: DatasourceTableLayout) => void;
  testId?: string;
  intl: IntlShape;
};

export interface LayoutButtonWrapperProps
  extends Pick<
    LayoutButtonProps,
    'mountPoint' | 'boundariesElement' | 'scrollableElement'
  > {
  api: ExtractInjectionAPI<typeof cardPlugin> | undefined;
  editorView: EditorView;
}
