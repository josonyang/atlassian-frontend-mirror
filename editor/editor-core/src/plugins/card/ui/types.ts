import { CardAppearance } from '@atlaskit/editor-common/provider-factory';

import { Command } from '@atlaskit/editor-common/types';

export interface OptionConfig {
  appearance?: CardAppearance;
  title: string;
  onClick: Command;
  selected: boolean;
  testId: string;
  disabled?: boolean;
  hidden?: boolean;
  tooltip?: string;
}
