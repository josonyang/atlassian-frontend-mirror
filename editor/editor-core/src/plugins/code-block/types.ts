import type { EditorAppearance } from '../../types';
import type { LongPressSelectionPluginOptions } from '@atlaskit/editor-common/types';

export interface CodeBlockOptions extends LongPressSelectionPluginOptions {
  allowCopyToClipboard?: boolean;
  allowCompositionInputOverride?: boolean;
  // We are not using EditorProps['appearance'] here as CodeBlockOptions is used
  // inside the type EditorProps, and so doing so triggers a recursion issue in
  // extract-react-types
  appearance?: EditorAppearance | undefined;
}
