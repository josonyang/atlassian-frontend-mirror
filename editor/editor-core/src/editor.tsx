/** @jsx jsx */
import React from 'react';

import { jsx } from '@emotion/react';

import { ComposableEditor } from './composable-editor';
import useUniversalPreset from './presets/useUniversalPreset';
import type { EditorProps } from './types/editor-props';
import editorDeprecationWarnings from './utils/editorDeprecationWarnings';

export type {
	AllowedBlockTypes,
	Command,
	CommandDispatch,
	DomAtPos,
	EditorAppearance,
	EditorAppearanceComponentProps,
	EditorConfig,
	EditorInstance,
	EditorPlugin,
	EditorProps,
	ExtensionConfig,
	ExtensionProvidersProp,
	MarkConfig,
	MessageDescriptor,
	NodeConfig,
	NodeViewConfig,
	PluginsOptions,
	PMPlugin,
	PMPluginCreateConfig,
	PMPluginFactory,
	PMPluginFactoryParams,
	ReactComponents,
	ToolbarUIComponentFactory,
	ToolbarUiComponentFactoryParams,
	UIComponentFactory,
	UiComponentFactoryParams,
} from './types';
export type { FeedbackInfo } from '@atlaskit/editor-common/types';

interface WrapperProps {
	props: EditorProps;
}

const ComposableEditorWrapper = ({ props }: WrapperProps) => {
	const preset = useUniversalPreset({ props });
	return <ComposableEditor preset={preset} {...props} />;
};

export default class Editor extends React.Component<EditorProps> {
	static defaultProps: EditorProps = {
		appearance: 'comment',
		disabled: false,
		extensionHandlers: {},
		allowHelpDialog: true,
		quickInsert: true,
	};

	constructor(props: EditorProps) {
		super(props);
		editorDeprecationWarnings(props);
	}

	render() {
		return <ComposableEditorWrapper props={this.props} />;
	}
}
