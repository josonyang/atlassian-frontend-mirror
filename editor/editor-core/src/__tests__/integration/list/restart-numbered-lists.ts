import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import {
  fullpage,
  editable,
  getDocFromElement,
  setProseMirrorTextSelection,
} from '@atlaskit/editor-test-helpers/integration/helpers';
import { selectors as editorSelectors } from '@atlaskit/editor-test-helpers/page-objects/editor';
import { WebDriverPage } from '@atlaskit/editor-test-helpers/integration/types';
import {
  goToEditorTestingWDExample,
  mountEditor,
} from '@atlaskit/editor-test-helpers/testing-example-page';

import emptyAdf from './__fixtures__/empty-adf.json';
import { createListAdf } from './__fixtures__/list-starting-from-adf';

const listStartingFrom3Adf = createListAdf({ order: 3 });
const listStartingFrom0Adf = createListAdf({ order: 0 });
const listStartingFrom1Point9Adf = createListAdf({ order: 1.9 });

const startEditor = async (client: any, adf: any): Promise<WebDriverPage> => {
  const page = await goToEditorTestingWDExample(client);
  await mountEditor(page, {
    appearance: fullpage.appearance,
    defaultValue: adf,
    featureFlags: {
      restartNumberedLists: true,
    },
  });
  return page;
};

BrowserTestCase(
  'restart-numbered-lists.ts: typing "99." inserts ordered list starting from 99',
  { skip: [] },
  async (client: any, testName: string) => {
    const page = await startEditor(client, emptyAdf);
    await page.keys([...`99.`.split(''), 'Space']);
    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'restart-numbered-lists.ts: typing "0." inserts ordered list starting from 0',
  { skip: [] },
  async (client: any, testName: string) => {
    const page = await startEditor(client, emptyAdf);
    await page.keys([...`0.`.split(''), 'Space']);
    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'restart-numbered-lists.ts: typing "-1." should not insert ordered list',
  { skip: [] },
  async (client: any, testName: string) => {
    const page = await startEditor(client, emptyAdf);
    await page.keys([...`-1.`.split(''), 'Space']);
    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'restart-numbered-lists.ts: typing "99." and extra text and pressing enter should continue incrementing from 99',
  { skip: [] },
  async (client: any, testName: string) => {
    const page = await startEditor(client, emptyAdf);
    // we use page.keys because windows firefox doesn't execute space chars correctly during page.type
    // (when the space is positioned between other chars) and also suffers issues finding the starting
    // selectors ("element <p> not interactable" errors)
    await page.keys([
      ...`99.`.split(''),
      'Space',
      'a',
      'Enter',
      'b',
      'Enter',
      'c',
      'Enter',
    ]);
    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'restart-numbered-lists.ts: typing "99." above an existing ordered list should join them and start from 99',
  { skip: [] },
  async (client: any, testName: string) => {
    const page = await startEditor(client, listStartingFrom3Adf);
    await page.click(editorSelectors.firstEditorParagraph);
    await page.keys([...`99.`.split(''), 'Space', 'x']);
    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'restart-numbered-lists.ts: typing "99." below an existing ordered list should join them and start from the existing list start number',
  { skip: [] },
  async (client: any, testName: string) => {
    const page = await startEditor(client, listStartingFrom3Adf);
    await page.click(editorSelectors.lastEditorChildParagraph);
    await page.keys([...`99.`.split(''), 'Space', 'x']);
    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'restart-numbered-lists.ts: splitting a list that starts from 3 (by pressing Enter) should create 2 lists that continue number sequence',
  { skip: [] },
  async (client: any, testName: string) => {
    const page = await startEditor(client, listStartingFrom3Adf);
    // set cursor to end of first list item
    await setProseMirrorTextSelection(page, { anchor: 6 });
    // and then split the list into two
    await page.keys(['Enter', 'Enter']);
    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'restart-numbered-lists.ts: splitting a list that starts from 0 (by pressing Enter) should create 2 lists that continue number sequence',
  { skip: [] },
  async (client: any, testName: string) => {
    const page = await startEditor(client, listStartingFrom0Adf);
    // set cursor to end of first list item
    await setProseMirrorTextSelection(page, { anchor: 6 });
    // and then split the list into two
    await page.keys(['Enter', 'Enter']);
    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'restart-numbered-lists.ts: splitting a list that starts from 1.9 (by pressing Enter) should create 2 lists that continue number sequence',
  { skip: [] },
  async (client: any, testName: string) => {
    const page = await startEditor(client, listStartingFrom1Point9Adf);
    // set cursor to end of first list item
    await setProseMirrorTextSelection(page, { anchor: 6 });
    // and then split the list into two
    await page.keys(['Enter', 'Enter']);
    // we expect a list of 1.9 and another list starting from 2 (since the browser would
    // render 1.9 as 1)
    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
