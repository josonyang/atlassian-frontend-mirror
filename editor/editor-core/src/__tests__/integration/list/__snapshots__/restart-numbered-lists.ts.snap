// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`restart-numbered-lists.ts: splitting a list that starts from 0 (by pressing Enter) should create 2 lists that continue number sequence 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "order": 0,
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "a",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "orderedList",
    },
    Object {
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "order": 1,
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "b",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "orderedList",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`restart-numbered-lists.ts: splitting a list that starts from 1.9 (by pressing Enter) should create 2 lists that continue number sequence 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "order": 1.9,
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "a",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "orderedList",
    },
    Object {
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "order": 2,
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "b",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "orderedList",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`restart-numbered-lists.ts: splitting a list that starts from 3 (by pressing Enter) should create 2 lists that continue number sequence 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "order": 3,
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "a",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "orderedList",
    },
    Object {
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "order": 4,
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "b",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "orderedList",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`restart-numbered-lists.ts: typing "-1." should not insert ordered list 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "-1. ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`restart-numbered-lists.ts: typing "0." inserts ordered list starting from 0 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "order": 0,
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "orderedList",
    },
  ],
  "type": "doc",
}
`;

exports[`restart-numbered-lists.ts: typing "99." above an existing ordered list should join them and start from 99 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "order": 99,
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "x",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "a",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "b",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "orderedList",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`restart-numbered-lists.ts: typing "99." and extra text and pressing enter should continue incrementing from 99 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "order": 99,
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "a",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "b",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "c",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "orderedList",
    },
  ],
  "type": "doc",
}
`;

exports[`restart-numbered-lists.ts: typing "99." below an existing ordered list should join them and start from the existing list start number 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "order": 3,
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "a",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "b",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "x",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "orderedList",
    },
  ],
  "type": "doc",
}
`;

exports[`restart-numbered-lists.ts: typing "99." inserts ordered list starting from 99 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "order": 99,
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "orderedList",
    },
  ],
  "type": "doc",
}
`;
